sdk/communication/Azure.Communication.PhoneNumbers/src/MSUserAgentPolicy.cs

```diff
@@ -42,6 +42,8 @@ public override async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory
         /// <param name="message"></param>
         private static void addHeader(HttpMessage message)
         {
+            message.Request.Headers.Add("x-ms-useragent", "acs-test");
```

Comment:
I am not clear on this change, why are we setting default header value? Wouldn't it be better to just set environment variable AZURE_USERAGENT_OVERRIDE?
Btw. for live runs in pipeline, this variable is retrieved from kevault, and I believe it already has this value.

Reply:
How is this only done in tests? Isn't this in the src/ directory?

If environment variable can't be found, we can't assume that it's from the test SDK. It could be from a misconfiguration.

Reply:
This class is internal class and is only used in ClientOptionsExtension, which is also internal and not exposed to the SDK users. 
Anyway, after the discussion with @jiriburant  I have abandoned the hardcoded value. Seems like Live tests already use environment variable override so there is no need to set it in the code. The only thing is, in case there is a need to re-generate test recordings, those calls won't be marked as acs-test but that case is rear so it should be ok.

 Please have another look. Thank you for your comments and review.